### PR TITLE
Add module example to T.self_type

### PR DESCRIPTION
**Pull request by qaisjp**
### Motivation

Someone at work asked a question that could have been solved by pointing to this docs page with this example.

However this may not be a desirable addition, as it's not too dissimilar to the existing example.

### Test plan

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20strict%0Aextend%20T%3A%3ASig%0A%0Aclass%20Parent%0A%20%20extend%20T%3A%3ASig%0A%0A%20%20sig%20%7Breturns(T.self_type)%7D%0A%20%20def%20foo%0A%20%20%20%20self%0A%20%20end%0Aend%0A%0Aclass%20Child%20%3C%20Parent%3B%20end%0A%0AT.reveal_type(Parent.new.foo)%20%23%20Revealed%20type%3A%20Parent%0AT.reveal_type(Child.new.foo)%20%23%20Revealed%20type%3A%20Child%0A%0Amodule%20Mixin%0A%20%20extend%20T%3A%3ASig%0A%20%20%0A%20%20sig%20%7Breturns(T.self_type)%7D%0A%20%20def%20bar%0A%20%20%20%20self%0A%20%20end%0Aend%0A%0Aclass%20UsesMixin%0A%20%20extend%20Mixin%0Aend%0A%0AT.reveal_type(UsesMixin.bar)%20%23%20Revealed%20type%3A%20T.class_of(UsesMixin))

### CODE REVIEW

File: website/docs/self-type.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ we call a method on i.e., `x` in `x.foo`). For instance, `#dup` returns
 type.
 
 ```ruby
-# typed: true
+# typed: strict
```
Comment (froydnj) [inline website/docs/self-type.md:22]: Our docs are not quite consistent, but I think the prevailing style is to use `# typed: true`; please revert this change.

**Review by froydnj — CHANGES_REQUESTED** [review]
I think this is a helpful clarification...but it looks like there are some linter issues in CI?

File: website/docs/self-type.md (hunk 1 above)
Comment (qaisjp) [reply website/docs/self-type.md:22]:
```suggestion
# typed: true
```

**Comment by qaisjp** [comment]
The lint error says: `^ To fix these files, run 'yarn prettier' in the website/ folder locally.`

So I checked out the branch, then ran these lines:

```
$ cd website

$ yarn
yarn install v1.22.4
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
warning Your current version of Yarn is out of date. The latest version is "1.22.5", while you're on "1.22.4".
info To upgrade, run the following command:
$ curl --compressed -o- -L https://yarnpkg.com/install.sh | bash
✨  Done in 10.03s.

$ yarn prettier
yarn run v1.22.4
$ prettier --write './**/*.js' './**/*.css' './**/*.md'
core/Footer.js 61ms
core/PageSection.js 9ms
pages/en/community.js 42ms
pages/en/index.js 68ms
siteConfig.js 27ms
static/js/error-reference.js 16ms
static/css/custom.css 94ms
static/css/overrides.css 19ms
static/css/PageSection.css 21ms
blog/2019-05-16-state-of-sorbet-spring-2019.md 181ms
blog/2019-06-20-open-sourcing-sorbet.md 57ms
blog/2019-12-20-announcing-sorbet-0.5.md 167ms
blog/2020-07-30-ruby-3-rbs-sorbet.md 72ms
docs/abstract.md 44ms
docs/adopting.md 59ms
docs/attached-class.md 38ms
docs/class-of.md 20ms
docs/class-types.md 42ms
docs/cli.md 56ms
docs/error-reference.md 224ms
docs/exhaustiveness.md 49ms
docs/faq.md 97ms
docs/final.md 54ms
docs/flow-sensitive.md 28ms
docs/gradual.md 57ms
docs/intersection-types.md 17ms
docs/legal/trademark-policy.md 91ms
docs/metrics.md 69ms
docs/nilable-types.md 51ms
docs/non-forcing-constants.md 27ms
docs/noreturn.md 12ms
docs/override-checking.md 29ms
docs/overview.md 20ms
docs/procs.md 52ms
docs/quickref.md 37ms
docs/rbi.md 77ms
docs/requires-ancestor.md 22ms
docs/runtime.md 81ms
docs/sealed.md 46ms
docs/self-type.md 13ms
docs/shapes.md 15ms
docs/sigs.md 69ms
docs/static.md 73ms
docs/stdlib-generics.md 20ms
docs/talks/curry-on-2019.md 19ms
docs/talks/jvm-ls-2019.md 15ms
docs/talks/ruby-conf-2019.md 11ms
docs/talks/ruby-kaigi-2018.md 20ms
docs/talks/ruby-kaigi-2019.md 13ms
docs/talks/strange-loop-2018.md 16ms
docs/tconfiguration.md 33ms
docs/tenum.md 73ms
docs/troubleshooting.md 47ms
docs/tstruct.md 14ms
docs/tuples.md 11ms
docs/type-aliases.md 17ms
docs/type-annotations.md 40ms
docs/type-assertions.md 37ms
docs/union-types.md 24ms
docs/untyped.md 31ms
style-guide.md 30ms
✨  Done in 3.57s.
```

What to do?

**Comment by qaisjp** [comment]
Doh. It modified files!

**Review by froydnj — APPROVED** [review]
Thank you!
